Accept trailing whitespace in case of using the Slack's slash commands

Caroga opened the pull request:
Slack slash command, for example `/cfp`, will be send with a trailing slash. I expanded the regex to cover a optional whitespace.

Slack will send slash commands with a trailing slash (`/cfp` will be received as `/cfp `) which causes the regex to not validate correctly.

mpociot commented:
Can you create a test that checks for a message that works when this gets applied?
This way we ensure, that this doesn't break in future releases.

Caroga commented:
Sure 👍

Caroga commented:
@mpociot I've added a basic test method for `Matcher.php` which checks several types of messages:
- messages containing no trailing whitespaces but whitespaces in between
- messages containing trailing whitespaces
- messages containing named parameters and no trailing whitespaces
- messages containing named parameters and trailing whitespaces

I've also added common prefixed messages, e.g. `/command` or `!command`.

I might have gone overboard with the amount of messages, but we can always remove something if needed. 

Would this suffice? 

P.s.: rebased on 2.0, should I squash commit this for you ?

mpociot commented:
That's great @Caroga  - thank you!

Caroga commented:
Glad to be of help.
